internal/epoch/epoch_manager.go

```diff
@@ -717,6 +771,29 @@ func (e *Manager) refreshAttemptLocked(ctx context.Context) error {
 	return nil
 }
 
+// MaybeAdvanceWriteEpoch writes a new write epoch marker when a new write
+// epoch should be started, otherwise it does not do anything.
+func (e *Manager) MaybeAdvanceWriteEpoch(ctx context.Context) error {
+	if !e.allowWrites() {
+		return nil
+	}
+
+	p, err := e.getParameters()
+	if err != nil {
+		return err
+	}
+
+	e.mu.Lock()
+	cs := e.lastKnownState
+	e.mu.Unlock()
+
+	if shouldAdvance(cs.UncompactedEpochSets[cs.WriteEpoch], p.MinEpochDuration, p.EpochAdvanceOnCountThreshold, p.EpochAdvanceOnTotalSizeBytesThreshold) {
+		return errors.Wrap(e.advanceEpochMarker(ctx, cs), "error advancing epoch")
+	}
```

Comment:
I don't recall this 100%, did we decide that we should advance epoch during maintenance only or that will continue to happen during index loads on write connection ?

In other words, with this change `e.allowWrites()` will skip epoch advance here by-default now since `allowWrites()` is going to be returning `false` always until someone overrides the index manager config from CLI.

Reply:
And I just noticed that `MaybeAdvanceWriteEpoch()` is exposed to be called from maintenance only. That makes sense, but what if maintenance is enabled and `e.allowWrites()` returns false ? Would that ever happen, if not is that check needed ?

Reply:
good catch, changed.